nni/retiarii/strategy/base.py

```diff
@@ -4,12 +4,38 @@
 import abc
 from typing import List
 
+from .. import report_search_space
 from ..graph import Model
 from ..mutator import Mutator
 
 
 class BaseStrategy(abc.ABC):
 
+    @classmethod
+    def report_model_space(self, base_model: Model, applied_mutators: List[Mutator]) -> None:
+        """
+        NOTE: nested search space is not supported (currently also not supported on webui),
+        currently, only support type `choice`.
+        """
+        search_space = {}
+        auto_label_seq = 0
+        new_model = base_model
+        for mutator in applied_mutators:
+            recorded_candidates, new_model = mutator.dry_run(new_model)
+            # TODO: check label is never None
+            if mutator.label is not None:
+                label = mutator.label
+            else:
+                label = f'auto_label_{auto_label_seq}'
```

Comment:
Where does `auto_label` come from? I think this logic is already done in `dry_run_for_search_space`, and `self` is never used in this function. Suggest refact it into a util.

Reply:
moved to util. `dry_run_for_search_space` cannot be used as the generated search space should follow our search space format